src/utility/Rfc822Address.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * [RFC 822](https://www.ietf.org/rfc/rfc822.txt) defines how email is
+ * structured and allows for the emails of the form 'cbothner@umich.edu' as well
+ * as the form 'Cameron Bothner <cbothner@umich.edu>'. This class parses and
+ * unpacks that format.
+ *
+ * @providesModule Rfc822Address
+ * @flow
+ */
+
+export default class Rfc822Address {
+  // Let’s try to match angle brackets no matter how they’re escaped. The truth
+  // is that the Gmail API gives us \u003c and \u003e.
+  static ANGLE_BRACKET_PATTERNS = [/<|\\u003c|&lt;/, />|\\u003e|&gt;/];
+
+  /* eslint-disable max-len */
+  // This is what the W3C says web browsers should use to validate emails in
+  // <input type="email"> elements. Sigh...
+  // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#Basic_validation
+  static BARE_EMAIL_ADDRESS_PATTERN = /[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*/;
```

Comment:
Holy crap, lol.

Reply:
I might steal this for next week's task 👍

Reply:
omg X_X I'm not even gonna pretend I understand this

Reply:
does this function help to get rid of the `\u003...` in the subject?